pkg/api/job_spec.go

```diff
@@ -17,13 +19,27 @@ type JobSpec struct {
 	rawSpec string
 
 	// these fields allow the job to be targeted at a location
-	Namespace     string
+	namespace     string
 	BaseNamespace string
 
 	// if set, any new artifacts will be a child of this object
 	owner *meta.OwnerReference
 }
 
+// Namespace returns the namespace of the job. Must not be evaluated
+// at step construction time because its unset there
+func (s *JobSpec) Namespace() string {
+	if s.namespace == "" {
+		log.Println("Warning, namespace accessed before it was set, this is a bug in ci-operator. Stack:")
+		debug.PrintStack()
```

Comment:
Hrm, should we just panic here? Something's likely going to fail later anyway and failing fast would force us to fix it,

Reply:
I thought about this but decided against it because it might not be a reason to abort the test. For example the place where I initially walked on this was that we failed to label the namespace to prevent it from getting deleted.

If it is a reason to fail the test (like with the ci-operator bug the other day) then the test will presumably fail anyways down the line.